File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.ts
function maxProfit(prices: number[]): number {
    var l = 0, r = 1;
    var maxProfit = 0;
    while(r < prices.length){
        if(prices[l] < prices[r]){
            var profit = prices[r] - prices[l];
            if(maxProfit <= profit){
                maxProfit = profit;
            }
        } else{
            l = r;
        }
        r++;
    }
    return maxProfit;
};

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        minProfit = float(inf)
        maxProfit = 0

        for price in prices:
            if minProfit > price:
                minProfit = price

            profit = price - minProfit

            if maxProfit < profit:
                maxProfit = profit
            
        return maxProfit        
